spring-boot/admin/src/test/java/eu/coatrack/admin/service/GatewayHealthMonitorServiceTest.java

```diff
@@ -0,0 +1,131 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.when;
+
+public class GatewayHealthMonitorServiceTest {
+
+    @Mock
+    ProxyRepository proxyRepository;
+
+    @Mock
+    Proxy proxySample;
+
+    private List<Proxy> sampleProxies = new ArrayList<>();
+
+    /*
+     * 5 and 60 minutes represent the threshold for the gateway health status.
+     * Warning state is triggered when the last call from a gateway was done more than 5 minutes ago, while Critical
+     * state will be triggered if was done more than 60 minutes ago
+     */
+    @InjectMocks
+    private GatewayHealthMonitorService gatewayHealthMonitorService = new GatewayHealthMonitorService(5, 60, proxyRepository);
+
+    public Proxy createSampleProxy(boolean monitoredEnabled, String proxyName) {
+        Proxy proxy = new Proxy();
+        proxy.setName(proxyName);
+        proxy.setId("testIdFor-" + proxyName);
+        proxy.setHealthMonitoringEnabled(monitoredEnabled);
+        proxy.updateTimeOfLastSuccessfulCallToAdmin_setToNow();
+        return proxy;
+    }
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        sampleProxies = Arrays.asList(createSampleProxy(true, "testProxyMonitoringEnabledA"),
+                createSampleProxy(false, "testProxyMonitoringDisabledA"),
+                proxySample);
+        when(proxyRepository.findAvailable()).thenReturn(sampleProxies);
+        when(proxySample.isHealthMonitoringEnabled()).thenReturn(true);
+        when(proxySample.getName()).thenReturn("testProxyMonitoringEnabledB");
+    }
+
+    @Test
+    public void ifAllGatewaysAreOkOrIgnore_ThenStatusSummaryShouldReturnOkState() {
+        when(proxySample.getTimeOfLastSuccessfulCallToAdmin()).thenReturn(LocalDateTime.now());
+        ProxyHealthStatus healthStatusSummary = gatewayHealthMonitorService
+                .calculateGatewayHealthStatusSummary(gatewayHealthMonitorService
+                        .getGatewayHealthMonitorData()
+                        .healthDataForAllGateways);
+        Assert.assertEquals(ProxyHealthStatus.OK, healthStatusSummary);
+    }
+
+    // The 6 means that the last gateway call was made 6 minutes ago, which should trigger a Warning State
+    @Test
+    public void ifOneGatewayIsWarningAndOthersOKOrIgnore_ThenStatusSummaryShouldReturnWarningState() {
+        when(proxySample.getTimeOfLastSuccessfulCallToAdmin()).thenReturn(LocalDateTime.now().minusMinutes(6));
+        ProxyHealthStatus healthStatusSummary = gatewayHealthMonitorService
+                .calculateGatewayHealthStatusSummary(gatewayHealthMonitorService
+                        .getGatewayHealthMonitorData()
+                        .healthDataForAllGateways);
+        Assert.assertEquals(ProxyHealthStatus.WARNING, healthStatusSummary);
+    }
+
+    // The 6 means that the last gateway call was made 70 minutes ago, which should trigger a Critical State
```

Comment:
```suggestion
    // The 70 means that the last gateway call was made 70 minutes ago, which should trigger a Critical State
```
and please move it down, as it is not a method-level comment but a line-level comment

Reply:
I also moved this comment on top of the line of code where the 70 is